### PR TITLE
[Hub Generated] Review request for Microsoft.AzureStackHCI to add version preview/2023-09-01-preview

### PR DESCRIPTION
**Pull request by pathuang-sia**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/pathuang-sia-azurestackhci-Microsoft.AzureStackHCI-2023-09-01...main/).
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by pathuang-sia** [comment]
Will reopen PR once e2e validation against RPSaaSMaster succeeds.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:9]: The x-ms-code-generation-setting extension is being deprecated. Please remove it and move settings to readme file for code generation.
Location: Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json#L7

Comment (TimLovellSmith) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:9]: I'm not sure if this is really a blocker, seems like you were already flagged for this with the old api version. Any hints, @raych1 ?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/marketplaceGalleryImages.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/marketplaceGalleryImages.json:224]: >OK [](http://example.com/codeflow?start=28&length=2)

Should perhaps say something a little more context specific like, 'The update succeeded?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/networkInterfaces.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/networkInterfaces.json:427]: >     "x-ms-client-flatten": true, [](http://example.com/codeflow?start=9&length=33)

nit - is this x-ms-client-flatten even meaningful on the property level when its not type: object? I don't think it is?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1087]: >VirtualMachineVMConfigAgentInstanceView [](http://example.com/codeflow?start=5&length=39)

This type name seems to have the similar double VM redundancy thing going on - want to fix it?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1531]: > "x-ms-secret": true, [](http://example.com/codeflow?start=9&length=21)

I'm surprised they are secret. Is this because some proxies are configured with authentication in the URL?

Is there some readonly property you'll expose instead to allow people to understand when a proxy is actually secretly configured?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1530]: >type [](http://example.com/codeflow?start=11&length=4)

should it be format url?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1548]: >connecting to proxy [](http://example.com/codeflow?start=57&length=19)

connecting to is a little vague on how the cert is used - can you clarify the description, like is this configuring the trusted CA for verifying proxy server certs? or something else?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/examples/PutLogicalNetwork.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/examples/PutLogicalNetwork.json:38]: >name [](http://example.com/codeflow?start=11&length=4)

the example value looks like a full resource id, so I would normally have expected this property is called 'id', not 'name'.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/examples/PutLogicalNetwork.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/examples/PutLogicalNetwork.json:39]: >CustomLocation [](http://example.com/codeflow?start=19&length=14)

seeing this brings up questions for me like, is it meant to imply the same type as
'Microsoft.ExtendedLocation/customLocations'
?
is it redundant, if its already part of the full resource id?
should it be camelCased?
Or are you PascalCasing it because you think of it more like an enum?
Is the enum going to be defined by AzureStackHCI, or who?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:87]: >: "azure-async-operation" [](http://example.com/codeflow?start=27&length=25)

I thought this is not right, for apis using azure async operation header - I think they are supposed to leave this blank, or use 'resource' - indicating the original URI is used to fetch the final state of the updated resource?

File: specification/azurestackhci/resource-manager/readme.md
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/readme.md:64]: Looking at this old suppression, I don't think its supposed to be needed. Can we remove it?

> code: GetCollectionOnlyHasValueAndNextLink
    reason: The linter is mistakenly thinking that paths for a singular resource that is always named default, like "/{resourceUri}/providers/Microsoft.AzureStackHCI/virtualMachineInstances/default," is for a collection of resources.

File: specification/azurestackhci/resource-manager/readme.md
Comment (TimLovellSmith) [inline specification/azurestackhci/resource-manager/readme.md:67]: Again I hope this suppression shouldn't be needed.

I wonder if the leading slash before '{resourceUri}': i.e. '/{resourceUri}' i the swagger paths, might be the actual cause of some of these linter errors.

Comment (TimLovellSmith) [reply specification/azurestackhci/resource-manager/readme.md:67]: (I can understand in this case we probably don't need list by subscription though.)

File: specification/azurestackhci/resource-manager/readme.md
Comment (TimLovellSmith) [reply specification/azurestackhci/resource-manager/readme.md:64]: I wonder if the leading slash before '{resourceUri}': i.e. '/{resourceUri}' i the swagger paths, might be the actual cause of some of these linter errors.

Does removing that make any difference?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/marketplaceGalleryImages.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/marketplaceGalleryImages.json:224]: We modeled against compute for this: https://github.com/Azure/azure-rest-api-specs-pr/blob/de06c42bd985fb66b6a49c907aaf5baee693bc7b/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-07-01/image.json#L168

Won't fix.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1531]: @djshah19 would you mind help answering this?

Comment (djshah19) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1531]: @TimLovellSmith that's right, some proxies are configured with authentication in the URL

File: specification/azurestackhci/resource-manager/readme.md
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/readme.md:64]: @wyatt-troia Does it make any difference without the leading slash?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (djshah19) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1548]: It is purely for client-side trust of the proxy. It is low value, it is not used for authentication.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1087]: Confirmed that it indeed has a redundancy. Will fix.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1531]: marking it as won'tfix. Feel free to reopen the conversation if there are lingering questions.
#won'tfix

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (TimLovellSmith) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1548]: Ask is please make the description clear.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/networkInterfaces.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/networkInterfaces.json:427]: will fix

Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/networkInterfaces.json:427]: yep makes sense to remove. Will fix

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/examples/PutLogicalNetwork.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/examples/PutLogicalNetwork.json:39]: It is defined as an enum here: https://github.com/Azure/azure-rest-api-specs-pr/blob/980cb0ae2dce82d554fd0925eb16dacad675d92f/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/common.json#L42

Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/examples/PutLogicalNetwork.json:39]: Yes since it was defined the same way for [scvmm](https://github.com/Azure/azure-rest-api-specs-pr/blob/2c81cf086a071cde7497c04c946c2985bf249dd5/specification/scvmm/resource-manager/Microsoft.ScVmm/preview/2023-04-01-preview/examples/CreateVirtualMachine.json) and [vmware](https://github.com/Azure/azure-rest-api-specs-pr/blob/RPSaaSMaster/specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2023-03-01-preview/examples/CreateVirtualMachine.json) as well, which all fall under the same arc resource model. I'll mark it as won'tfix similar to the reason from previous comment. 

#won'tfix

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:87]: This was according to guideline here: https://azure.github.io/autorest/extensions/#x-ms-long-running-operation-options that RPaaS requires us to do so. These are async operations, which azure supports both location and azure-async-operation for final states. We use azure-async-operation which is the newer option. 

#won'tfix

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1548]: will fix

File: specification/azurestackhci/resource-manager/readme.md
Comment (wyatt-troia-msft) [reply specification/azurestackhci/resource-manager/readme.md:64]: I don't know. I never tried that

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json:1530]: will fix

File: specification/azurestackhci/resource-manager/readme.md
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/readme.md:67]: The linter actually indicates that the property is not allowed without the leading slash. We will keep this suppression for now, as the cluster resources require this suppression as well. 

Let me know if you would like to revisit it.

#won'tfix

File: specification/azurestackhci/resource-manager/readme.md
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/readme.md:64]: I gave it a try locally, and the linter actually doesn't allow for paths without a leading slash.

Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/readme.md]: suppression removed. Fixed.

**Comment by pathuang-sia** [comment]
@konrad-jamrozik Hey Konrad, we would like to request an approval for avocado as we do not carry the resources mentioned in this api-version and are not modifying them.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json]: fixed

**Comment by konrad-jamrozik** [comment]
> @konrad-jamrozik Hey Konrad, we would like to request an approval for avocado as we do not carry the resources mentioned in this api-version and are not modifying them.

@pathuang-sia Avocado is not a blocking check anymore. It is up to the PR merger (see step 4 in https://aka.ms/azsdk/pr-diagram) to determine if it precludes you from merging, or not.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (raosuhas) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:499]: why is the etag called as ID ? Is there a mismatch between the name and description ?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (raosuhas) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:503]: >Resource [](http://example.com/codeflow?start=44&length=8)

did you mean to add the           "readOnly": true, for this ?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (raosuhas) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:512]: >Resource [](http://example.com/codeflow?start=44&length=8)

smae here ? Maybe the routetable itself should be readonly ?

**Comment by raosuhas** [comment]
Signed off , with a few minor comments. Please fix them before merging.

**Comment by raych1** [comment]
@pathuang-sia , can you share the reason of removing `arcSettings.json`,`clusters.json`, etc. in this api version while comparing to previous stable version? This causes the breaking change in SDK.

**Comment by pathuang-sia** [comment]
> @pathuang-sia , can you share the reason of removing `arcSettings.json`,`clusters.json`, etc. in this api version while comparing to previous stable version? This causes the breaking change in SDK.

Hi @raych1, the Microsoft.AzureStackHCI resource provider is shared between the cluster team and our team that owns the arc vm scenario/resources. We are in different release stages and therefore have agreed that we won't carry their resources in this api-version.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:499]: Yep ideally should be named etag. I will mark it as a won'tfix for this api-version however since the property description is still accurate enough to avoid user misunderstanding but is a property name change that will trigger bottom stack changes. Let me know if you think otherwise. 

#won'tfix

**Comment by pathuang-sia** [comment]
Hi @msyyc, would it be possible to get the CI-BreakingChange-Python-Track2 approval?

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (TimLovellSmith) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:499]: So this being a stable API version, you're not going to be able to rename it in the future without it being a breaking change. You should probably fix it now or never. @pathuang-sia.

Comment (TimLovellSmith) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:499]: What I guess you *could* do, in lieu of a breaking change, is add an additional etag property and mark this old property deprecated, to make your API easier to undersand in the future, without breaking existing customers. But why not just avoid such a mess altogether?

File: specification/azurestackhci/resource-manager/readme.md
Comment (TimLovellSmith) [reply specification/azurestackhci/resource-manager/readme.md:64]: Really? I thought we had a linter rule saying the reverse. I'm officially confused! /cc @rkmanda

File: specification/azurestackhci/resource-manager/readme.md
Comment (TimLovellSmith) [reply specification/azurestackhci/resource-manager/readme.md]: Unresolving pending below discussion

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json:499]: Yeah it makes more sense to do it right now. I spoke with the scenario owner and we will fix the property name in this api-version.

Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json]: fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json]: fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/logicalNetworks.json]: fixed

**Comment by pathuang-sia** [comment]
Note: we have renamed the api-version to a preview version under the guidance of the breaking change committee

**Comment by raych1** [comment]
The breaking change for Go is not brought in by this PR.

**Comment by ms-zhenhua** [comment]
please fix the error code in lint-diff https://github.com/Azure/azure-rest-api-specs/pull/25555/checks?check_run_id=17531976023, then re-add MergeRequested

**Comment by pathuang-sia** [comment]
@ms-zhenhua We believe the DefinitionPropertiesNameCamelCase error is false positive. The rule indicates that if it is a two letter acronym it should be capitalized like how we did: https://github.com/Azure/azure-openapi-validator/blob/main/docs/definitions-properties-names-camel-case.md

Can we bypass this error and merge the PR?

**Comment by ms-zhenhua** [comment]
> @ms-zhenhua We believe the DefinitionPropertiesNameCamelCase error is false positive. The rule indicates that if it is a two letter acronym it should be capitalized like how we did: https://github.com/Azure/azure-openapi-validator/blob/main/docs/definitions-properties-names-camel-case.md
> 
> Can we bypass this error and merge the PR?

@pathuang-sia I think you are right. Could you kindly follow [Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) to suppress **DefinitionsPropertiesNamesCamelCase** to fix **Automated merging requirements met** ? After that, please re-add **MergeRequested** label for merging. Thanks.

**Review by ms-zhenhua — APPROVED** [review]

**Pull request merged**
